.stylelintrc.json

```diff
@@ -0,0 +1,12 @@
+{
+  "extends": "stylelint-config-sass-guidelines",
+  "rules": {
+    "selector-class-pattern": [
+      "^[a-zA-Z0-9]+(?:-[a-zA-Z0-9]+)*(?:__[a-zA-Z0-9]+(?:-[a-zA-Z0-9]+)*){0,1}(?:(?:--)[a-zA-Z0-9]+(?:-[a-zA-Z0-9]+)*){0,1}$",
```

Comment:
This regex is based on [this plugin](https://github.com/postcss/postcss-bem-linter/blob/master/lib/preset-patterns.js). Breaking it in chunks:

- block: `[a-zA-Z0-9]+(?:-[a-zA-Z0-9]+)*`
- element: `(?:__[a-zA-Z0-9]+(?:-[a-zA-Z0-9]+)*){0,1}`
- modifier: `(?:(?:--)[a-zA-Z0-9]+(?:-[a-zA-Z0-9]+)*){0,1}`

Both element and modifier can only have one of them per class, e.g `nav__list__item` is not allowed in BEM convention.

Reply:
![image](https://user-images.githubusercontent.com/13206817/83091509-b49a9680-a071-11ea-9374-04afd316aef8.png)